manifests/09-prometheus-rules.yaml

```diff
@@ -5,8 +5,31 @@ metadata:
   namespace: openshift-image-registry
   annotations:
     capability.openshift.io/name: ImageRegistry
-    release.openshift.io/delete: "true"
     include.release.openshift.io/hypershift: "true"
     include.release.openshift.io/ibm-cloud-managed: "true"
     include.release.openshift.io/self-managed-high-availability: "true"
     include.release.openshift.io/single-node-developer: "true"
+spec:
+  groups:
+    - name: pvc-problem-detector.rules
+      rules:
+      - alert: ImageRegistryStorageReadOnly
+        for: 10m
+        expr: imageregistry_storage_errors_total {code="READ_ONLY_FILESYSTEM"}
```

Comment:
The metric is indeed a counter ([source](https://github.com/openshift/image-registry/blob/master/pkg/dockerregistry/server/metrics/prometheus.go?plain=1#L126-L134)).

> If the metric is a counter, you need a rate() (or increase()) otherwise the alert will never go back inactive when the problem is fixed.

This makes sense. With your suggestion will the alert reset after 5m of the metric not being present? I'm still a bit confused by these time filters 😅 

> Otherwise the alert would reset everytime a new pod is rolled out.

Ah that explains why I saw the alert "fix itself" after I fixed the storage (which forced a new pod to be rolled out). This was a total hack - I'll incorporate your suggestion.

Reply:
Additionally, I've set `for` to `10m`, but the `expr` you suggest has `5m`. Should these times match? 🤔

Reply:
`for: 10m` means that the alert will go firing (meaning delivered to Alertmanager) only after being active for 10 minutes. The `[5m]` range is the interval on which the rate is applied.